src/main/java/appeng/util/item/AEItemStack.java

```diff
@@ -106,15 +106,25 @@ public static IAEItemStack fromNBT( final NBTTagCompound i )
 		return item;
 	}
 
-	public static IAEItemStack fromPacket( final ByteBuf data ) throws IOException
+	@Override
+	public void writeToNBT( final NBTTagCompound i )
+	{
+		this.getDefinition().writeToNBT( i );
+		i.setLong( "Cnt", this.getStackSize() );
+		i.setLong( "Req", this.getCountRequestable() );
+		i.setBoolean( "Craft", this.isCraftable() );
+	}
+
+	public static AEItemStack fromPacket( final ByteBuf data ) throws IOException
 	{
 		final byte mask = data.readByte();
-		// byte PriorityType = (byte) (mask & 0x03);
+		final byte IdType = (byte) ( mask & 0x03 );
 		final byte stackType = (byte) ( ( mask & 0x0C ) >> 2 );
 		final byte countReqType = (byte) ( ( mask & 0x30 ) >> 4 );
 		final boolean isCraftable = ( mask & 0x40 ) > 0;
 
 		final ItemStack itemstack = ByteBufUtils.readItemStack( data );
+		final long stackId = getPacketValue( IdType, data );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Move the declaration of "stackId" closer to the code that uses it. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1941)